client/jetpack-connect/controller.js

```diff
@@ -106,6 +107,11 @@ export function redirectWithoutLocaleIfLoggedIn( context, next ) {
 
 export function maybeOnboard( { query, store }, next ) {
 	if ( ! isEmpty( query ) && query.onboarding ) {
+		if ( versionCompare( query.jp_version, '5.8', '<' ) ) {
+			// page.redirect() doesn't work here because of the different domain
+			return window.location.replace( query.site_url + '/wp-admin/' );
```

Comment:
Could we use `{ externalRedirect } from 'lib/route';` instead of `window.location.replace` here?

Also, we should be careful about using query args without checking if they exist, because they're coming from a query string after all... Should we protect the `query.site_url` usage with a `get()`?